Reconnect on TimeoutError

Artimi opened the pull request:
This should solve https://github.com/aio-libs/aiokafka/issues/144. When there is raised exception `asyncio.TimeoutError` from `AIOKafkaConnection` in `Client.send` it will delete connection from `Client._conns`. This means that next time we do some request to this node new connection is created and therefore we get rid of stale connection. There is still raised `KafkaTimeoutError` so you can retry your request if needed. I added a simple test for this. Please advise if there is something wrong with my code.

Drizzt1991 commented:
Thanks, few thoughts:

 * You need to close the socket before discarding connections, or you'll end up with stale sockets. I would recommend you just close the connection, rather than discarding it.
 * There could be multiple requests pipelined on this connection. Closing will probably raise ConnectionError in those, rather than KafkaTimeoutError, need to think if that's ok.

Drizzt1991 commented:
To check yourself for coverage, please run:
```
make cov
make diff-cov
```
It will show you how much of your diff is covered by tests.

Artimi commented:
Thanks for feedback. I changed it to close and added test that `close()` really make connection `not conn.connected()`. Problem with tests was that I omitted `@run_until_complete` decorator. Yes if I close connection with multiple requests in line it will raise `ConnectionError` but I guess this is right because I will have new connection then and requests might be retried.

Drizzt1991 commented on tests/test_conn.py:
```diff
@@ -204,3 +204,12 @@ def invoke_osserror(*a, **kw):
+
+    @run_until_complete
+    def test_close_disconnects_connection(self):
+        asyncio.set_event_loop(self.loop)
```
Don't do this in tests. It's not without a reason it's there, you probably didn't pass loop explicitly somewhere.

Artimi replied:
Ok sorry about that I was just expanding test `test_global_loop_for_create_conn` and this remained there.
